pkg/webhook/controlplane/ensurer.go

```diff
@@ -198,16 +198,17 @@ func ensureKubeSchedulerCommandLineArgs(c *corev1.Container, csiEnabled, csiMigr
 }
 
 func ensureKubeControllerManagerLabels(t *corev1.PodTemplateSpec, csiEnabled, csiMigrationComplete bool) {
+	// make sure to always remove this label
+	delete(t.Labels, v1beta1constants.LabelNetworkPolicyToBlockedCIDRs)
```

Comment:
I think this explicit removal is not required. I tested with the change I'm suggesting below and during reconciliation this label is being removed from the kube-controller-manager Pod.

```suggestion
```